typescript/packages/plugins/mode-governance/src/mode-governance.service.ts

```diff
@@ -0,0 +1,115 @@
+import { Tool } from "@goat-sdk/core";
+import { EVMWalletClient } from "@goat-sdk/wallet-evm";
+import { parseUnits, formatUnits } from "viem";
+import { VOTING_ESCROW_ABI, MODE_TOKEN_ABI, BPT_TOKEN_ABI } from "./abi";
+import {
+    MODE_TOKEN_ADDRESS,
+    BPT_TOKEN_ADDRESS,
+    MODE_VOTING_ESCROW,
+    BPT_VOTING_ESCROW,
+} from "./constants";
+import {
+    StakeParameters,
+    GetStakeInfoParameters,
+    GetBalanceParameters,
+} from "./parameters";
+
+export class ModeGovernanceService {
+    @Tool({
+        description: "Stakes MODE or BPT tokens in the Mode governance system",
+    })
+    async stakeTokens(walletClient: EVMWalletClient, parameters: StakeParameters) {
+        const tokenAddress = parameters.tokenType === "MODE" ? MODE_TOKEN_ADDRESS : BPT_TOKEN_ADDRESS;
+        const escrowAddress = parameters.tokenType === "MODE" ? MODE_VOTING_ESCROW : BPT_VOTING_ESCROW;
+        
+        const approveHash = await walletClient.sendTransaction({
```

Comment:
I would not do this in the same function. There is already an approve function in the erc20 plugin, the LLM should figure out it needs to use that first. I would add it to the description:

"Requires MODE or BPT tokens to be approved"

Reply:
got it , added it to the description rather, sounds good.